lib/kubernetes-deploy/options_helper.rb

```diff
@@ -2,25 +2,58 @@
 
 module KubernetesDeploy
   module OptionsHelper
-    def self.default_and_check_template_dir(template_dir)
-      if !template_dir && ENV.key?("ENVIRONMENT")
-        template_dir = "config/deploy/#{ENV['ENVIRONMENT']}"
+    class OptionsError < StandardError; end
+
+    STDIN_TEMP_FILE = "from_stdin.yml.erb"
+    class << self
+      def with_consolidated_template_dir(template_dirs)
+        if template_dirs.length <= 1 && ARGV.empty? && $stdin.tty?
+          yield default_template_dir(template_dirs.first)
+        else
+          Dir.mktmpdir("kubernetes-deploy") do |dir|
+            populate_temp_dir(temp_dir: dir, template_dirs: template_dirs)
+            yield dir
+          end
+        end
       end
 
-      if !template_dir || template_dir.empty?
-        puts "Template directory is unknown. " \
-          "Either specify --template-dir argument or set $ENVIRONMENT to use config/deploy/$ENVIRONMENT " \
-        + "as a default path."
-        exit(1)
+      def revision_from_environment
+        ENV.fetch('REVISION') do
+          puts "ENV['REVISION'] is missing. Please specify the commit SHA"
```

Comment:
Let's raise this as an exception instead of exiting outside the executable

Reply:
Handled by #429